src/Microsoft.ML.Core/Data/ReadOnlyMemoryUtils.cs

```diff
@@ -0,0 +1,397 @@
+                return -1;
+            if (len < a.Length)
+                return +1;
+            return 0;
+        }
+
+        public static IEnumerable<ReadOnlyMemory<char>> Split(char[] separators, ReadOnlyMemory<char> memory)
+        {
+            Contracts.CheckValueOrNull(separators);
+
+            if (memory.IsEmpty)
+                yield break;
+
+            if (separators == null || separators.Length == 0)
+            {
+                yield return memory;
+                yield break;
+            }
+
+            if (separators.Length == 1)
+            {
+                char chSep = separators[0];
+                for (int ichCur = 0; ;)
+                {
+                    int ichMinLocal = ichCur;
+                    for (; ; ichCur++)
+                    {
+                        Contracts.Assert(ichCur <= memory.Length);
+                        if (ichCur >= memory.Length)
+                        {
+                            yield return memory.Slice(ichMinLocal, ichCur - ichMinLocal);
+                            yield break;
+                        }
+                        if (memory.Span[ichCur] == chSep)
+                            break;
+                    }
+
+                    yield return memory.Slice(ichMinLocal, ichCur - ichMinLocal);
+
+                    // Skip the separator.
+                    ichCur++;
+                }
+            }
+            else
+            {
+                for (int ichCur = 0; ;)
+                {
+                    int ichMinLocal = ichCur;
+                    for (; ; ichCur++)
+                    {
+                        Contracts.Assert(ichCur <= memory.Length);
+                        if (ichCur >= memory.Length)
+                        {
+                            yield return memory.Slice(ichMinLocal, ichCur - ichMinLocal);
+                            yield break;
+                        }
+                        // REVIEW: Can this be faster?
+                        if (ContainsChar(memory.Span[ichCur], separators))
+                            break;
+                    }
+
+                    yield return memory.Slice(ichMinLocal, ichCur - ichMinLocal);
+
+                    // Skip the separator.
+                    ichCur++;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Splits <paramref name="memory"/> on the left-most occurrence of separator and produces the left
+        /// and right <see cref="ReadOnlyMemory{T}"/> of <see cref="char"/> values. If <paramref name="memory"/> does not contain the separator character,
+        /// this returns false and sets <paramref name="left"/> to this instance and <paramref name="right"/>
+        /// to the default <see cref="ReadOnlyMemory{T}"/> of <see cref="char"/> value.
+        /// </summary>
+        public static bool SplitOne(char separator, out ReadOnlyMemory<char> left, out ReadOnlyMemory<char> right, ReadOnlyMemory<char> memory)
```

Comment:
The order of parameters to this method seems wrong.  Typically the `out` parameters are at the end.